benchmarks/dynamo/pr_time_benchmarks/benchmark_base.py

```diff
@@ -64,29 +75,42 @@ def description(self):
         return ""
 
     @abstractmethod
-    def prepare(self):
+    def _prepare(self):
         pass
 
     @abstractmethod
-    def work(self):
+    def _work(self):
         pass
 
-    def prepare_once(self):  # noqa: B027
+    def _prepare_once(self):  # noqa: B027
         pass
 
-    def count_instructions(self):
+    def _count_instructions(self):
         print(f"collecting instruction count for {self.name()}")
-        self.prepare_once()
-
         results = []
         for i in range(10):
-            self.prepare()
+            self._prepare()
             id = i_counter.start()
-            self.work()
+            self._work()
             count = i_counter.end(id)
             print(f"instruction count for iteration {i} is {count}")
-            if i != 0:
-                results.append(count)
+            results.append(count)
+        return min(results)
+
+    def _count_compile_time_instructions(self):
```

Comment:
I'm confused as to the difference between `_count_instructions` and `_count_compile_time_instructions`.  Don't they do the same thing but one uses _instruction_counter directly and the other indirectly?

Reply:
# measure total number of instruction spent in _work.
    _enable_instruction_count = False

    # measure total number of instruction spent in convert_frame.compile_inner
    # TODO is there other parts we need to add ?
    _enable_compile_time_instruction_count = False

Reply:
# measure total number of instruction spent in _work.
 _enable_instruction_count = False 
 
 # measure total number of instruction spent in convert_frame.compile_inner # TODO is there other parts we need to add ? _enable_compile_time_instruction_count = False

Reply:
if your program  is a micro beanchmark measure some function in isolate. you would use _enable_instruction_count and not _enable_compile_time_instruction_count because
_enable_compile_time_instruction_count would return 0.